src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -90,33 +91,55 @@ export function CurrencySearch({
     return Object.values(defaultTokens).filter(getTokenFilter(debouncedQuery))
   }, [defaultTokens, debouncedQuery])
 
-  const [balances, balancesAreLoading] = useAllTokenBalances()
+  const { data, loading: balancesAreLoading } = usePortfolioBalancesQuery({
+    skip: !account,
+    variables: { ownerAddress: account ?? '' },
+    fetchPolicy: 'cache-only', // PrefetchBalancesWrapper handles balance fetching/staleness; this component only reads from cache
+    errorPolicy: 'all',
+  })
```

Comment:
maybe since we're duplicating this code now across 3 different places, we should just export a hook like `usePrefetchedBalances()` from `PrefetchedBalancesWrapper`